### PR TITLE
add docker support

### PR DESCRIPTION
**Pull request by kandros**
production:
```
docker-compose build
```
```
docker-compose up
```

development
```
docker-compose docker-compose-development.yml build
```
```
docker-compose docker-compose-development.yml up
```

> Mac osx users must change webpack host to docker-machine ip

#155

### CODE REVIEW

**Comment by carloscuatin** [comment]
this for production true ?

**Comment by carloscuatin** [comment]
@kandros somehow keep this in development?

**Comment by kandros** [comment]
@carloscuatin both added

**Comment by carloscuatin** [comment]
@kandros there is no problem with webpack or react hot loader or anything right?

**Comment by kandros** [comment]
should work out of the box on linux, on other OS the file sync from the host to the virtual machine running docker should be configured manually with rsync or similar tools

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+IFROM node
```
Comment (somus) [inline Dockerfile]: I think it should be `FROM` instead of `IFROM`.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+web:
+  build: .
+  links:
+   - db
+  ports:
+   - "8080:8000"
```
Comment (somus) [inline docker-compose.yml]: Here the port should be exposed at `8000` instead of `8080` or was it intentional?

**Comment by somus** [comment]
@kandros Awesome work! 👍 

I've pointed out few changes. Take look at those. Also, it would be awesome if you add the same how to use Docker in the README file. I'll merge once those are completed.

File: docker-compose.yml (hunk 2 above)
Comment (kandros) [inline docker-compose.yml]: 80 was intended

**Comment by joernroeder** [comment]
cool! can't wait to play with it 👍

**Comment by somus** [comment]
@kandros Merged. Thanks for PR!

**Comment by rickisen** [comment]
I'm admittedly new to Docker and Mern, but should the mongo container really
allow connections from outside the docker network or localhost by default?

If I run docker-compose with this docker-compose.yml on a vps, I open up the db for anyone to log on.
I guess this could work in development, but isn't docker-compose.yml meant for production?

Wouldn't it be better to change to:
```
db:
  image: mongo:latest
  ports:
   - "127.0.0.1:27017:2701
```
So it only accepts from localhost

**Pull request merged**